### PR TITLE
Update xdma for purecap.

### PR DESCRIPTION
**Pull request by qwattash**

### CODE REVIEW

File: sys/dev/altera/jtag_uart/altera_jtag_uart_cons.c [hunk 1]
```diff
@@ -108,48 +116,61 @@ SYSCTL_UINT(_hw_altera_jtag_uart, OID_AUTO, ac_poll_delay,
...
+static inline char *
+mips_aju_register_addr(vm_paddr_t regoff)
 {
 
-	return (phys | MIPS_XKPHYS_UNCACHED_BASE);
+#ifdef __CHERI_PURE_CAPABILITY__
+	return (cheri_setoffset(aju_io_capability, regoff));
+#else
+	return (MIPS_PHYS_TO_XKPHYS_UNCACHED(BERI_UART_BASE + regoff));
+#endif
```
Comment (jrtc27) [inline sys/dev/altera/jtag_uart/altera_jtag_uart_cons.c:132]: Maybe `aju_io_capability` should be an `aju_io_pointer` that gets set to BERI_UART_BASE for !purecap, and then we just use normal pointer arithmetic?

Comment (qwattash) [reply sys/dev/altera/jtag_uart/altera_jtag_uart_cons.c:132]: good point

File: sys/dev/altera/jtag_uart/altera_jtag_uart_cons.c [hunk 2]
```diff
@@ -69,6 +72,11 @@ int		aju_cons_jtag_present;
 u_int		aju_cons_jtag_missed;
```
Comment (jrtc27) [inline sys/dev/altera/jtag_uart/altera_jtag_uart_cons.c:72]: This driver is full of sadness with global state and no use of bus functions... not your problem to solve, but maybe we'll want/need something more mature for the DE10-Pro.

File: sys/dev/xdma/xdma.h [hunk 3]
```diff
@@ -95,19 +95,24 @@ struct xchan_buf {
...
 	uint32_t			nsegs_left;
-	vm_offset_t			vaddr;
+	vm_pointer_t			vaddr;
 	vm_offset_t			paddr;
 	vm_size_t			size;
 };
 
+typedef union {
+	uintptr_t		vaddr;
+	bus_addr_t		addr;
```
Comment (jrtc27) [inline sys/dev/xdma/xdma.h:105]: These are somewhat confusing names...

Comment (qwattash) [reply sys/dev/xdma/xdma.h:105]: Perhasp `vaddr` and `bus_addr` are clearer.

Comment (qwattash) [reply sys/dev/xdma/xdma.h:105]: Although this means that probably we want to change the naming also in bus_dma.h as it somewhat mirrors that one.

File: sys/dev/xilinx/axidma.c [hunk 4]
```diff
@@ -364,11 +364,11 @@ axidma_desc_alloc(struct axidma_softc *sc, struct xdma_channel *xchan,
 	pmap_kenter_device(chan->mem_vaddr, chan->mem_size, chan->mem_paddr);
 
 	device_printf(sc->dev, "Allocated chunk %lx %lu\n",
-	    chan->mem_paddr, chan->mem_size);
+	    (vaddr_t)chan->mem_paddr, chan->mem_size);
```
Comment (brooksdavis) [inline sys/dev/xilinx/axidma.c:367]: ptraddr_t

**Review by brooksdavis — APPROVED** [review]
One stray vaddr_t.

**Pull request merged**